Python/remote_debugging.c

```diff
@@ -595,10 +633,7 @@ read_memory(proc_handle_t *handle, uint64_t remote_address, size_t len, void* ds
     }
     return 0;
 #else
-    PyErr_SetString(
-        PyExc_RuntimeError,
-        "Memory reading is not supported on this platform");
-    return -1;
+    Py_UNREACHABLE();
```

Comment:
Does this do something if it is actually reached? I'd guess `Py_FatalError` is the only real option, since it doesn't know how to return an error code.

Reply:
They are equivalent. This one crashes with the funny message from xkcd:

https://github.com/python/cpython/blob/275056a7fdcbe36aaac494b4183ae59943a338eb/Include/pymacro.h#L152-L179

Reply:
I don't know about the others, but it's not runtime-safe for execution to actually reach `__assume(0)` - it implies genuine lack of reachability, and UB if it's reached (which probably crashes, but it's not guaranteed). Most of the time we won't get the funny messages.

It shouldn't matter in this case, because we have Windows code everywhere you've used it, but we should more clearly decide how we want to handle `Py_UNREACHABLE`.

Reply:
I think this is the case, we should only reach this I we really screwed up the guards

Reply:
Maybe we can blow up at compile time?